test/dhtest.c

```diff
@@ -511,6 +514,8 @@ static int run_rfc5114_tests(void)
 
         Z1 = OPENSSL_malloc(DH_size(dhA));
         Z2 = OPENSSL_malloc(DH_size(dhB));
+        if ((!Z1) || (!Z2))
+            goto bad_err;
```

Comment:
This could leak memory if it fails (e.g. if `Z2` malloc fails then `dhA`, `dhB` and `Z2` could leak. Same with all the `bad_err` and `err` gotos in this function.